Force a config volume to be mapped, and change how passwords are set

coppit opened the pull request:
If you don't like these changes I can release my own image.

coppit committed:
Update host networking note

It's not necessary to use host networking. I ran it fine with bridged networking.

coppit committed:
Clean up the run command a bit, fixing a typo.

coppit committed:
Expand the description to help the search engine.

coppit committed:
Bump the version of Phusion to the latest

coppit committed:
Speed up apt, and exit on error

coppit committed:
Dropping the firstrun command

Doesn't look like it's being used...

coppit committed:
Change how configuration and startup is handled

Force the user to set the password in a config file before the container runs "for real". This forces them to use a
config volume, which avoids the situation where they might accidentally invalidate their license.

I also added some verbiage about how it's a bit unclear how the licensing works. On the assumption that it's tied to the
MAC address of the internet-facing NIC, I'm storing that in case it's needed later.

FWIW, I have emailed their support staff to get some clarifications. For instance they say that the Lite version
prohibits it running in a VM, but I don't know if that applies to containers too.

coppit committed:
Remove upgrade of all packages.

Speeds up the build. Plus I'd rather let Phusion update the base packages.

jedimonkey commented:
I'll try to check this out on the weekend!

jedimonkey commented:
Hey @coppit, Docker builds fine, but it doesn't appear to running xeoma properly, as I can't connect to it remotely.  Only started digging through your changes a bit, but couldn't see anything obvious.  Will keep having a look.  I tried just rebuilding my beta version (so it uses the latest Xeoma stable and beta binaries) and it seemed to build/run fine, so I don't believe it's anything to do with Xeoma changing anything.  One thing I did notice is you purposedly get the mac address and set a config, which I'm not sure is working for me.  Can you send me the docker run string you are starting it with?

coppit commented on Dockerfile:
```diff
@@ -13,7 +13,7 @@ RUN echo "force-unsafe-io" > /etc/dpkg/dpkg.cfg.d/02apt-speedup \
 
 # Install prerequisites
 RUN set -x && \
-  apt-get update && apt-get -y upgrade && \
```
From https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/: "You should avoid RUN apt-get upgrade or dist-upgrade"

jedimonkey replied:
you miss what I meant, I mean what command are you starting the container with?  I run mine with 
```
docker run -d --restart=always --name xeoma --net=host -v /etc/localtime:/etc/localtime:ro -v /share/docker/xeoma:/usr/local/Xeoma -v /mnt/archive:/usr/local/Xeoma/XeomaArchive jedimonkey/xeoma:beta
```

jedimonkey replied:
I don't fully agree with the 'don't run apt-get upgrade' as it means basically run (potentially) with old libraries.  From a security perspective, it's potentially dangerous, and relying on updates from upstream.  Doing a dist-upgrade is silly though to do in docker though.

coppit replied:
Yeah, I wanted to call this out because I could see you wanting to keep it in. Feel free to drop that part.

coppit commented:
> Can you send me the docker run string you are starting it with?

See the gory details below. I'll take a look and see if there's anything special in there. I'm running it non-privileged, bridged networking, opening up port 8090.

/plugins/dynamix.docker.manager/scripts/docker create --name="Xeoma" --net="bridge" -e TZ="America/New_York" -e HOST_OS="unRAID" -p 8090:8090/tcp -v "/mnt/cache/appdata/xeoma":"/usr/local/Xeoma":rw -v "/mnt/disks/Hitachi_HDS5C3030ALA630_MJ1313YNG2450C/xeoma-archive":"/usr/local/Xeoma/XeomaArchive":rw coppit/xeoma-server

coppit commented:
BTW, if the readme command doesn't work, we should figure out why:

sudo docker run -d --name=xeoma --restart=always -p 8090:8090 -v /local/path/to/config:/usr/local/Xeoma jedimonkey/xeoma

coppit commented:
> One thing I did notice is you purposedly get the mac address and set a config

I save the MAC, but don't use it for anything. I'm just being overly cautious. I mention in the README that it _might_ be necessary to set it, but it hasn't been necessary for me. I think that as long as you have an internet connection, Xeoma can validate your license just fine.

I'm not seeing any gotchas. Let me know if you're still having trouble after looking at my run command. Be sure that you're mapping /usr/local/Xeoma in the container to the same location that you have in the past.

jedimonkey commented:
Interesting, you are able to use bridge networking, where as I'm using host.  I do recall trying to get this to work with bridge, but none of the camera detection worked for me.  That may have changed now.  Will try to test it more later tonight.

coppit commented:
Oh! I specified my cameras manually. Let me test some more and update the
readme about that.

We're you able to connect?

On Mon, Apr 3, 2017 at 6:48 AM Marcus Collier <notifications@github.com>
wrote:

> Interesting, you are able to use bridge networking, where as I'm using
> host. I do recall trying to get this to work with bridge, but none of the
> camera detection worked for me. That may have changed now. Will try to test
> it more later tonight.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jedimonkey/xeoma-docker/pull/2#issuecomment-291109348>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAe0WQr-1stcKQEK4OQhEB0T0_IRJ4_eks5rsM6JgaJpZM4MsdmR>
> .
>

coppit committed:
Enhance the README

Add info about bridged networking affecting camera searching, and on how changing networking will cause a licensing
error.

coppit commented:
I tried switching to host networking, and I got an error about licensing. I don't want to relicense just to test this, so I'll take your word that the cameras can't be found.

I pushed an update to the README to talk about the camera search issue, how to get relicensed if you choose to change the networking, and more. But for some reason it's not getting added to this pull request. I'm kinda a git noob about this stuff. If you want to grab it, it's here: https://github.com/coppit/docker-xeoma-server/blob/master/README.md

captainsdc commented:
Hi Marcus/Coppit.  I am trying to get this docker running on my QNAP nas (running linux 3.4.6 under the hood) and having issues.  Is there anything either of you could offer me to try to get the initial settings configured to run it in Container Station?  I am too noob to do much under the command line, but I am very persistent... Scott
